### ISSUE TITLE
Make MicroElements.Swashbuckle.FluentValidation less host and generator dependent

### ISSUE THREAD

**Issue by petriashev** [micro-elements/MicroElements.Swashbuckle.FluentValidation#63]
There are many requests to support other hosts (AzureFunctions) or other generators (NSwag).
Package should be splitted to several parts.

Supposed parts:

- Common logic
- Swachbuckle generator support
- NSwag generator support
- AspNetCore hosting model
- Azure Functions hosting

**Comment by ChillerShock** [comment]
Is it possible to make this package work without registering FluentValidation in `Startup.cs`?
Registering FluentValidation leads to following issues (correct me if it's wrong):

1. You are not able to use async validations
Because to use them you have to call `ValidateAsync` method on validator object.
2. You are not able to get `ValidationResult.Errors` (to me they are better than standard ASP.NET Core error response)

**Comment by buvinghausen** [comment]
@ChillerShock you are correct the built in MVC pipeline does everything synchronously therefore it is ultimately not a good choice for FluentValidation for anything but very rudimentary options.

The root of the issue is this library is dependent upon a IValidatorFactory instance which doesn't happen if you don't use the FluentValidation.AspNetCore package.

Just add the following line and you'll be good to go...

```cs
services.AddTransient<IValidatorFactory, HttpContextServiceProviderValidatorFactory>();
```

**Comment by buvinghausen** [comment]
@petriashev I agree on all fronts.... this package is too opinionated as currently constructed

**Comment by petriashev** [comment]
Started refactoring toward this issue
